modules/unsupported/csv/src/main/java/org/geotools/data/csv/parse/CSVSpecifiedWKTStrategy.java

```diff
@@ -42,18 +52,68 @@ public CSVSpecifiedWKTStrategy(CSVFileState csvFileState, String wktField) {
 
     @Override
     protected SimpleFeatureType buildFeatureType() {
-        SimpleFeatureTypeBuilder builder = CSVStrategySupport.createBuilder(csvFileState);
+    	// sets up a builder with a good guess at attribtues and values
+    	// (based on trying to parse the first line of content to detect numbers, dates, etc...)
+        SimpleFeatureTypeBuilder builder = CSVStrategy.createBuilder(csvFileState);
         String[] csvHeaders = csvFileState.getCSVHeaders();
         List<String> headers = Arrays.asList(csvHeaders);
-        if (headers.contains(wktField)) {
-            builder.remove(wktField);
-            builder.add(wktField, Geometry.class);
+        int index = headers.indexOf(wktField);
+        if(index != -1) {
+        	builder.remove(wktField);
+        	
+        	AttributeTypeBuilder builder2 = new AttributeTypeBuilder();
+        	builder2.setCRS(DefaultGeographicCRS.WGS84);
+        	builder2.binding(Geometry.class);       	
+        	AttributeDescriptor descriptior = builder2.buildDescriptor(wktField);
+			builder.add(index, descriptior);
         }
         return builder.buildFeatureType();
     }
-
+    
+    
     @Override
-    public SimpleFeature createFeature(String recordId, String[] csvRecord) {
+    public void createSchema(SimpleFeatureType featureType) throws IOException {
+        List<String> header = new ArrayList<String>();
+      
+        for (AttributeDescriptor descriptor : featureType.getAttributeDescriptors()) {
+            if (descriptor instanceof GeometryDescriptor) {
+            	header.add(wktField);
+            } else {
+            	header.add(descriptor.getLocalName());
+            }
+        }
+        // Write out header, producing an empty file of the correct type
+        CsvWriter writer = new CsvWriter(new FileWriter(this.csvFileState.getFile()),',');
+        try {
+            writer.writeRecord( header.toArray(new String[header.size()]));
+        }
+        finally {
+            writer.close();
+        }
+    }
+    
+    @Override
+    public String[] encode(SimpleFeature feature) {
+    	List<String> csvRecord = new ArrayList<String>();
+        for (Property property : feature.getProperties()) {
+        	String name = property.getName().getLocalPart();
+            Object value = property.getValue();
+            if (value == null) {
+            	csvRecord.add("");
+            } else if (name.compareTo(wktField) == 0) {
+            	WKTWriter wkt = new WKTWriter();            	
```

Comment:
We probably have a custom GeoTools WKTWriter that can do curves ... this is just a nice to have.